package.json

```diff
@@ -100,6 +102,8 @@
     "karma-jasmine": "3.1.0",
     "karma-jasmine-html-reporter": "1.5.1",
     "npm-run-all": "4.1.5",
+    "prettier": "^2.2.1",
```

Comment:
> Would there be a need to disable certain rules in our TSLint config, so that they do not conflict with prettier?

I took a look at the [conflicting rules](https://unpkg.com/tslint-config-prettier@1.18.0/lib/index.json) and it seems to suggest that a large portion of our TSLint config is formatting rules, which will be redundant if we choose to use Prettier.

I decided to leave the rules in just because Prettier does not throw errors if the code does not adhere to said style.

This also ties in with why Prettier should be run as a hook, so that changes can be guaranteed to adhere to style. Although we can choose another way of running Prettier to avoid adding another dependency.

Reply:
> Do we have to use an exact version of prettier?
> From the [Prettier docs](https://prettier.io/docs/en/install.html#summary):
> 
> > Install an exact version of Prettier locally in your project. This makes sure that everyone in the project gets the exact same version of Prettier. Even a patch release of Prettier can result in slightly different formatting, so you wouldn’t want different team members using different versions and formatting each other’s changes back and forth.

True.. I'll fix the version at `2.2.1` (latest stable at point of writing)

Reply:
> I took a look at the conflicting rules and it seems to suggest that a large portion of our TSLint config is formatting rules, which will be redundant if we choose to use Prettier.

I ran the `tslint-config-prettier-check` tool described [here](https://github.com/prettier/tslint-config-prettier#cli-helper-tool), and it only identified the following tslint rules as conflicting:
```
eofline
import-spacing
indent
max-line-length
no-trailing-whitespace
one-line
quotemark
semicolon
typedef-whitespace
whitespace
```
Let me know if I missed something here.

> I decided to leave the rules in just because Prettier does not throw errors if the code does not adhere to said style.

My concern is whether the two tools (Prettier and tslint) might give opposing recommendations, creating inconveniences for our CI.
However, since the number of conflicting rules is small, this shouldn't be too tough to manage.
I think we can go ahead and integrate prettier.not
If we find conflicting recommendations, we can tweak the tslint rules.

>   I propose that we first adopt Prettier for the tests and e2e folders, and subsequently for src.
>> I think that's reasonable! Shall I create a new issue for that?

Instead of a new issue, can we setup prettier for `tests` and `e2e` within this PR?
We can setup `.prettierignore` to exclude all other files.
In future, we can adopt prettier for the rest of the codebase.

@ptvrajsk can we also get your thoughts on this ?
(best way to integrate prettier's standardised code formatting into our project)

Reply:
> My concern is whether the two tools (Prettier and tslint) might give opposing recommendations, creating inconveniences for our CI.

Oh yeah, I've already tweaked the prettierrc file so that Prettier will follow the existing tslint rules haha. Prettier will not cause any processes to fail at all, if that's what you mean

> Instead of a new issue, can we setup prettier for tests and e2e within this PR?

Yup, let me clarify this means that we want to apply prettier formatting on the files in these folders, not to restrict the automatic formatting hook to those folders right? (or is it both?)

Reply:
> Oh yeah, I've already tweaked the prettierrc file so that Prettier will follow the existing tslint rules haha

Ok, got it :+1: 

> Yup, let me clarify this means that we want to apply prettier formatting on the files in these folders, not to restrict the automatic formatting hook to those folders right? (or is it both?)

Both